openedx/core/djangoapps/user_authn/api/form_fields.py

```diff
@@ -1,16 +1,63 @@
 """
 Field Descriptions
 """
+from django import forms
 from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
 from django.utils.translation import gettext as _
 
 from common.djangoapps.student.models import UserProfile
 from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
 from openedx.core.djangoapps.user_api import accounts
 from openedx.core.djangoapps.user_authn.api.constants import SUPPORTED_FIELDS_TYPES
 
+FIELD_TYPE_MAP = {
+    forms.CharField: "text",
+    forms.PasswordInput: "password",
+    forms.ChoiceField: "select",
+    forms.TypedChoiceField: "select",
+    forms.Textarea: "textarea",
+    forms.BooleanField: "checkbox",
+    forms.EmailField: "email",
+}
 
-def _add_field_with_configurable_select_options(field_name, field_label, error_message=''):
+
+def add_extension_form_field(field_name, custom_form, field_description, field_type='required'):
```

Comment:
field_type's default value won't be required if we set a default value to `FIELD_TYPE` constant